dropshot/src/server.rs

```diff
@@ -327,6 +332,39 @@ impl<C: ServerContext> InnerHttpServerStarter<C> {
     }
 }
 
+/// This accepts connections more gracefully from a TcpListener.
+struct HttpAcceptor {
```

Comment:
Now that we're incorporating this pretty tricky functionality into Dropshot, we should probably have some test coverage for it.  Any ideas how to do that?  It'd be really nice if we could exercise some of those error cases but that seems hard.

Reply:
I did not have a way to programmatically trigger those error kinds. Maybe if it was sufficiently generic, such that some `ErroringListener` could be provided. But that won't be available yet.

Reply:
Okay.  How about the happy path?  What about a test that starts a Dropshot server and makes 100 or 1000 connections to it in a row?  Just thinking out loud about how best to test this automatically.